### PR TITLE
Update delta-related schema fields, tests, and validation

### PR DESCRIPTION
**Pull request by jayrbolton**
- Update ncbi taxon schemas to have delta fields (from, to, id)
- Update ncbi taxon tests to work with these fields
- Update the schema validator to more specifically validate delta vs non-delta schemas
- Update GO schemas to have the "delta: true" flag for validation

### CODE REVIEW

File: schemas/GO/GO_edges_consider.yaml [hunk 1]
```diff
@@ -1,6 +1,7 @@
----
 name: GO_edges_consider
```
Comment (MrCreosote) [inline schemas/GO/GO_edges_consider.yaml]: Note there's only 1 GO edges collection at this point.

Out of scope, but the GO and ENVO schemas are not exactly aligned to what the delta loader is producing

File: schemas/ncbi/ncbi_child_of_taxon.yaml [hunk 2]
```diff
@@ -1,15 +1,17 @@
 name: ncbi_child_of_taxon
 type: edge
+delta: true
+
 schema:
   "$schema": http://json-schema.org/draft-07/schema#
   type: object
-  required: [_from, _to, child_type]
+  required: [from, to, child_type]
```
Comment (MrCreosote) [inline schemas/ncbi/ncbi_child_of_taxon.yaml]: add `id`

File: test/stored_queries/test_ncbi_tax.py [hunk 3]
```diff
@@ -59,18 +71,18 @@ def setUpClass(cls):
             {'_from': 'ws_object_version/1:1:2', '_to': 'ncbi_taxon/1', 'assigned_by': 'assn2'},
             {'_from': 'ws_object_version/2:1:1', '_to': 'ncbi_taxon/1', 'assigned_by': 'assn2'},
         ]
-        ws_docs = [{'_key': '1', 'is_public': True}, {'_key': '2', 'is_public': False}]
+        ws_docs = [{'_key': '1', 'is_public': True}, {'_key': '1', 'is_public': False}]
```
Comment (MrCreosote) [inline test/stored_queries/test_ncbi_tax.py]: Not sure if this change was intentional? If so, what's the purpose?

File: schemas/ncbi/ncbi_child_of_taxon.yaml (hunk 2 above)
Comment (MrCreosote) [reply schemas/ncbi/ncbi_child_of_taxon.yaml]: Wait, there's no `child_type` field, we took that out a while ago

File: test/stored_queries/test_ncbi_tax.py (hunk 3 above)
Comment (jayrbolton) [reply test/stored_queries/test_ncbi_tax.py]: Good catch thanks

**Comment by jayrbolton** [comment]
Re: schemas matching current data loading -- this is a job for a future PR

**Pull request merged**